providers/openshift_deploy_metrics.rb

```diff
@@ -52,8 +47,11 @@ def whyrun_supported?
   end
 
   execute 'Deploy Cluster Metrics' do
-    command "#{node['cookbook-openshift3']['openshift_common_client_binary']} new-app --template=metrics-deployer-template --as=system:serviceaccount:openshift-infra:metrics-deployer #{new_resource.metrics_params.map { |opt, value| " -p #{opt}=#{value}" }.join(' ')} -n openshift-infra --config=admin.kubeconfig"
+    command "#{node['cookbook-openshift3']['openshift_common_client_binary']} new-app --template=metrics-deployer-template ${service_account} #{new_resource.metrics_params.map { |opt, value| " -p #{opt}=#{value}" }.join(' ')} -n openshift-infra --config=admin.kubeconfig"
     cwd node['cookbook-openshift3']['openshift_master_config_dir']
+    environment(
+      'service_account' => ose_major_version.split('.')[1].to_i.between?(3, 4) ? '--as=system:serviceaccount:openshift-infra:metrics-deployer' : ''
```

Comment:
this is clever way to support both versions with slightly different syntax ("--as")